experimental/PropertyDDS/packages/property-shared-tree-interop/src/schemaConverter.ts

```diff
@@ -0,0 +1,362 @@
+		}
+
+		const splitTypeId = TypeIdHelper.extractContext(referencedTypeId);
+		let typeSchema: NamedTreeSchema | undefined;
+
+		if (splitTypeId.context === "single") {
+			if (TypeIdHelper.isPrimitiveType(splitTypeId.typeid)) {
+				// @TODO for simplicity we convert it to native string
+				// if (splitTypeId.typeid === "String") {
+				//     // String is a special case, we actually have to represent it as a sequence
+				//     typeSchema = {
+				//             name: referencedTypeId,
+				//             localFields: new Map<LocalFieldKey, FieldSchema>([
+				//                 // TODO: What should be the key we use for the entries? Should this be standardized?
+				//                 ["entries" as LocalFieldKey, {
+				//                     kind: FieldKinds.sequence.identifier,
+				//                     types: new Set([
+				//                         // TODO: Which type do we use for characters?
+				//                     ]),
+				//                 }],
+				//             ]),
+				//             globalFields: new Set(),
+				//             extraLocalFields: emptyField,
+				//             extraGlobalFields: false,
+				//             value: ValueSchema.Nothing,
+				//         };
+				// } else {
+				let valueType: ValueSchema;
+				if (splitTypeId.isEnum) {
+					valueType = ValueSchema.Number;
+				} else if (
+					splitTypeId.typeid === "String" ||
+					splitTypeId.typeid.startsWith("Reference<") ||
+					splitTypeId.typeid === "Reference"
+				) {
+					valueType = ValueSchema.String;
+				} else if (booleanTypes.has(splitTypeId.typeid)) {
+					valueType = ValueSchema.Boolean;
+				} else if (numberTypes.has(splitTypeId.typeid)) {
+					valueType = ValueSchema.Number;
+				} else {
+					throw new Error(`Unknown primitive typeid: ${splitTypeId.typeid}`);
+				}
+
+				typeSchema = namedTreeSchema({
+					name: referencedTypeId,
+					extraLocalFields: emptyField,
+					value: valueType,
+				});
+				// }
+			} else {
+				if (splitTypeId.typeid === "NodeProperty") {
+					typeSchema = namedTreeSchema({
+						name: referencedTypeId,
+						extraLocalFields: fieldSchema(FieldKinds.optional),
+					});
+				} else {
+					const localFields = {};
+					const inheritanceChain = PropertyFactory.getAllParentsForTemplate(
+						splitTypeId.typeid,
+					);
+					inheritanceChain.push(splitTypeId.typeid);
+
+					for (const typeIdInInheritanceChain of inheritanceChain) {
+						if (typeIdInInheritanceChain === "NodeProperty") {
+							continue;
+						}
+
+						const schema = PropertyFactory.getTemplate(typeIdInInheritanceChain);
+						if (schema === undefined) {
+							throw new Error(
+								`Unknown typeid referenced: ${typeIdInInheritanceChain}`,
+							);
+						}
+						if (schema.properties !== undefined) {
+							for (const property of schema.properties) {
+								if (
+									property.properties &&
+									!isIgnoreNestedProperties(property.typeid)
+								) {
+									// TODO: Handle nested properties
```

Comment:
Should we throw here?

Reply:
Fixed in cc1cca2a8cf6cf2d54cb3ec380ff72fe708dd4a3